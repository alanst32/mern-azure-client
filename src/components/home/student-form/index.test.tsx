import React, { useState } from "react";
import {shallow} from 'enzyme';
import StudentForm from '.';
import setupMockStates from 'jest-react-hooks-mock';

describe('StudentForm', () => {
    let wrapper;
    const setState = jest.fn();
    const useStateMock: any = (initState: any) => [initState, setState];

    beforeEach(() => {
        wrapper = shallow(<StudentForm totalStudents={5}></StudentForm>);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('Render StudentForm properly', () => {
        expect(wrapper).toMatchSnapshot();
    });
})