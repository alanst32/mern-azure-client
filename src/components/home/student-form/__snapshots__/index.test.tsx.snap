// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentForm Render StudentForm properly 1`] = `
<div
  className="makeStyles-header-1"
>
  <JumbotronHeader
    totalStudents={5}
  />
  <form
    autoComplete="off"
    className="makeStyles-form-3"
    noValidate={true}
  >
    <WithStyles(ForwardRef(TextField))
      id="firstName"
      label="First Name"
      onChange={[Function]}
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      id="lastName"
      label="Last Name"
      onChange={[Function]}
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      id="country"
      label="Country"
      onChange={[Function]}
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      InputLabelProps={
        Object {
          "shrink": true,
        }
      }
      id="dateOfBirth"
      label="DOB"
      onChange={[Function]}
      type="date"
      variant="outlined"
    />
    <WithStyles(ForwardRef(Button))
      className="makeStyles-insertBtn-6"
      color="primary"
      id="insertBtn"
      onClick={[Function]}
      variant="contained"
    >
      Insert
    </WithStyles(ForwardRef(Button))>
  </form>
</div>
`;
